crates/uv-resolver/src/lock/mod.rs

```diff
@@ -4825,7 +4826,13 @@ fn normalize_requirement(
                     index.remove_credentials();
                     index
                 })
-                .map(|index| IndexMetadata::from(IndexUrl::from(VerbatimUrl::from_url(index))));
+                .map(|index| {
+                    IndexMetadata::from(IndexUrl::from(
+                        VerbatimUrl::from_url(index)
+                            .without_trailing_slash()
```

Comment:
We only currently call `VerbatimUrl::without_trailing_slash` in cases where we need the owned value, so it's not strictly necessary to return a `Cow`. Instead of `VerbatimUrl::from_url(index).without_trailing_slash().into_owned()` it could be something like `VerbatimUrl::from_url_normalized(index)`. I haven't made this change because, if we end up needing this elsewhere, returning a `Cow` could still be beneficial.